### PR TITLE
feat(users): Add `force_two_factor_auth` environment variable

### LINKED ISSUE #4774: feat: Add config to force 2FA for users
Currently 2FA is skippable in all the environments.

We need to force 2FA in prod later at some point, so there should be config to which forces 2FA is enabled.

### PR DESCRIPTION
**Pull request by ThisIsMani**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR will add force 2fa environment variable. When enabled it will force users to complete 2FA to access the control center.

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [x] This PR modifies application configuration/environment variables

`config/config.example.toml`
`config/deployments/integration_test.toml`
`config/deployments/production.toml`
`config/deployments/sandbox.toml`
`config/development.toml`
`config/docker_compose.toml`
`loadtest/config/development.toml`

## Motivation and Context

Closes #4774.

## How did you test it?

1. 2FA Status API
  - `is_skippable` field is added in the response.
    ```
    curl --location 'http://localhost:8080/user/2fa/v2' \
    --header 'Authorization: JWT' \
    ```
    ```
    {
        "status": null,
        "is_skippable": true
    }
    ```
2. Terminate 2FA API
  - This API will now not allow skip 2FA if the `force_two_factor_auth`
    ```
    curl --location 'http://localhost:8080/user/2fa/terminate?skip_two_factor_auth=true' \
    --header 'Authorization: JWT' \
    ```
    ```
    {
        "error": {
            "type": "invalid_request",
            "message": "Two factor auth required",
            "code": "UR_40"
        }
    }
    ```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by jarnura — DISMISSED** [review]

File: crates/router/src/core/user.rs [hunk 1]
```diff
@@ -1927,7 +1927,7 @@ pub async fn terminate_two_factor_auth(
         .change_context(UserErrors::InternalServerError)?
         .into();
 
-    if !skip_two_factor_auth {
+    if !skip_two_factor_auth || state.conf.user.force_two_factor_auth {
```
Comment (SanchithHegde) [inline crates/router/src/core/user.rs:1930]: Should it be the other way round: 

```suggestion
    if state.conf.user.force_two_factor_auth || !skip_two_factor_auth {
```

Considering the short-circuiting behavior of `||`, shouldn't `state.conf.user.force_two_factor_auth` have higher weightage over the `skip_two_factor_auth` query param?

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/router/src/core/user.rs (hunk 1 above)
Comment (ThisIsMani) [reply crates/router/src/core/user.rs:1930]: Yeah, config should be given more priority. I will change this.

**Review by Riddhiagrawal001 — DISMISSED** [review]

File: crates/router/src/core/user.rs (hunk 1 above)
Comment (ThisIsMani) [reply crates/router/src/core/user.rs]: Business logic wise what you said it correct.

This doesn't change the API in anyway. But I will change this though.

**Review by SanchithHegde — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by apoorvdixit88 — APPROVED** [review]

**Review by racnan — APPROVED** [review]

**Pull request merged**